### PR TITLE
fix(icon): twitter logo update in footer

### PR DESCRIPTION
**Pull request by preetsahil**
I have changed the twitter logo

### CODE REVIEW

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:1619]: This selector isn't specific enough, as it would affect all `.item` (menu item, dropdown item, etc).
The desired change should only affect the footer links so we should make the seector more specific
```css
.footer .ui.link.list > .item:hover svg g{
```

There also is a dedicated section inside docs.css for the footer adjustments (Line 29+) Please do your changes there instead of inside the code tag section for a better overview.

File: server/partials/footer.html
Comment (lubber-de) [inline server/partials/footer.html:25]: What's the source of the svg icon? Please provide a link. We need to check the usage license.

File: server/partials/footer.html
Comment (lubber-de) [inline server/partials/footer.html:29]: no need to comment out, you can remove it

File: server/partials/footer.html
Comment (lubber-de) [inline server/partials/footer.html:34]: No need to comment out, you can remove it

File: server/partials/footer.html
Comment (lubber-de) [inline server/partials/footer.html:37]: No need to comment out, you can remove it

File: package-lock.json
Comment (lubber-de) [inline package-lock.json:1]: Do'nt provide a package-lock.json. We switched to yarn. (I told you on discord already :wink:)

File: yarn.lock
Comment (lubber-de) [inline yarn.lock:7]: Discard all changes inside yarn.lock, it has nothing to do with your PR content (and, we switched to yarn registry anyway)

**Review by lubber-de — CHANGES_REQUESTED** [review]
Please see my comments.

File: server/files/stylesheets/docs.css
Comment (preetsahil) [reply server/files/stylesheets/docs.css:1619]: Okay let me make changes.

File: server/partials/footer.html
Comment (preetsahil) [reply server/partials/footer.html:25]: https://icons8.com/icons/set/twitter

Comment (lubber-de) [reply server/partials/footer.html:25]: There is an official brand logo provided by x.com, we should use that instead

https://about.twitter.com/en/who-we-are/brand-toolkit
https://about.twitter.com/content/dam/about-twitter/x/brand-toolkit/x-logo.zip

Comment (preetsahil) [reply server/partials/footer.html:25]: okay

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:43]: As stated below i suggest to use the original x brand SVG and style it using CSS as follows.
The CSS will also take care about the same transition and color behavior as for the other list items.

```suggestion
.footer .ui.link.list > .item svg {
    width: 0.7em;
    height: auto;
    margin-left: 0.25em;
    margin-right: 0.25em;
    fill: rgba(0,0,0,.4);
    transition: 0.1s fill ease;
}
.footer .ui.link.list > .item:hover svg {
  fill: rgba(0,0,0,.8);
}
```

File: server/partials/footer.html
Comment (lubber-de) [inline server/partials/footer.html:29]: That's still not the original brand logo (it does not have a square outline)
Please use the original brand svg and scale it accordingly via css (See suggested CSS Code above)

See differences:
First line is my suggested code using original brand icon from x.com website
Second is your PR 
image

```suggestion
<svg width="1200" height="1227" viewBox="0 0 1200 1227" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M714.163 519.284L1160.89 0H1055.03L667.137 450.887L357.328 0H0L468.492 681.821L0 1226.37H105.866L515.491 750.218L842.672 1226.37H1200L714.137 519.284H714.163ZM569.165 687.828L521.697 619.934L144.011 79.6944H306.615L611.412 515.685L658.88 583.579L1055.08 1150.3H892.476L569.165 687.854V687.828Z"></path>
</svg> Twitter
```

**Review by lubber-de — CHANGES_REQUESTED** [review]
Please see my comments again

**Comment by lubber-de** [comment]
> Maybe we should upgrade [Font Awesome](https://fontawesome.com/v6/docs/changelog/) library to use that new twitter X icon?

 That would add another 700 icons (and would mean to wait until a next FUI release)
We already had the discussion if/when to upgrade to FA 6.x at https://github.com/fomantic/Fomantic-UI/issues/2228#issuecomment-1033079137

**Comment by lubber-de** [comment]
Otherwise: Twitter/X has not even updated their follow/tweet/post embeded button widgets (which we use at the header), so i dont even think it's _that_ urgent to change the icon at the footer (now) (x is supposed to be a global brand and just redirects to twitter.com now.)
See the discussion the people at fontawesome had about it [here](https://github.com/FortAwesome/Font-Awesome/issues/19827)

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:51]: 2 spaces indentation please

File: server/partials/footer.html
Comment (lubber-de) [inline server/partials/footer.html:28]: No inline styling. Please enhance the css selector in docs.css instead.

```css
.footer .ui.link.list > .item svg+span {
  vertical-align: middle;
}
```

**Review by lubber-de — CHANGES_REQUESTED** [review]
We are close 😊 please see my comments again.

File: server/partials/footer.html
Comment (preetsahil) [reply server/partials/footer.html:28]: Actually with the previous twitter icon I have to do the align 
I will make the changes

Comment (ko2in) [reply server/partials/footer.html:28]: Should remove left margin from svg tag.

fui-docs-twitter-logo-update

Comment (ko2in) [reply server/partials/footer.html]: Sorry. Ignore this. The left margin makes the text align properly. I can't notice without zoom-in.

Comment (preetsahil) [reply server/partials/footer.html]: I have done all the changes that were told to be done.

File: server/files/stylesheets/docs.css
Comment (ko2in) [inline server/files/stylesheets/docs.css:44]: I think better use default `baseline` by removing this.
fui-doc-twitter-logo-align

Comment (preetsahil) [reply server/files/stylesheets/docs.css:44]: Yes it is looking good without using vertical-align.
I will change it

Comment (ko2in) [reply server/files/stylesheets/docs.css:44]: Please add also `margin-left: 0.25em` back. I was wrong about that and already mentioned it.

fui-doc-twitter-logo-update

File: server/partials/footer.html
Comment (phongha-trueanthem) [inline server/partials/footer.html:25]: should the icon be updated in FUI assets instead of manually inserting the new svg here?

Comment (preetsahil) [reply server/partials/footer.html:25]: Check the conversion above what we have discussed

Comment (ko2in) [reply server/partials/footer.html:25]: Yes. We will later upgrade to FontAwesome 6.x which has the new X brand icon. See the [comment](https://github.com/fomantic/Fomantic-UI-Docs/pull/479#issuecomment-1745510759) above.

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:45]: Please add `margin-left: 0.2em;`

#### Without any margin-left
image

#### With margin-left: 0.2em
image

**Review by lubber-de — CHANGES_REQUESTED** [review]
Just one step until ready for merge

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:44]: 0.3em should be sufficient, no trailing zeros needed

Comment (preetsahil) [reply server/files/stylesheets/docs.css:44]: Okay I going to change now

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:50]: Please add one empty line before a comment block

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:1619]: Please keep the empty line before a comment block

File: server/partials/footer.html
Comment (preetsahil) [reply server/partials/footer.html:29]: I think i added by mistake

File: server/files/stylesheets/docs.css
Comment (preetsahil) [reply server/files/stylesheets/docs.css]: corrected the mistakes you can look again please

File: server/files/stylesheets/docs.css
Comment (lubber-de) [inline server/files/stylesheets/docs.css:44]: Leave a space between key and value (yes, i am nitpicking, but so is prettier.io / linting
```suggestion
  margin-left: 0.2em;
```

**Review by lubber-de — APPROVED** [review]
LGTM. Thanks for your patience!

File: server/files/stylesheets/docs.css
Comment (preetsahil) [reply server/files/stylesheets/docs.css]: I have seen most of the open source project use prettier.

**Pull request merged**
